src/components/HOC/requireAuth.js

```diff
@@ -0,0 +1,25 @@
+import React, { Component } from 'react';
+import Cookies from 'universal-cookie';
+
+const cookies = new Cookies();
+export default ChildComponent => {
+  class ComposedComponent extends Component {
```

Comment:
@plxity Is that life cycle method actually needed?

Reply:
And I think a much more simple implementation can be done here
```
export default ProtectedRoute = props => {
       const isLoggedIn = !!cookies.get('loggedIn');
       if (!isLoggedIn) {
         this.props.history.push('/');
       }
       return (
             { props.children }
        )
}
```

and then wrapping the component in the index.js files without the Routes iteself

```
<Route path="/settings" render={()=>(<ProtectedRoute><Settings/></ProtectedRoute/>)}/>
```

@plxity Let me know if this works, this will simplify logic and make the code much more readable.

Reply:
@anshumanv Can you suggest which approach should I use?

Reply:
The component should not be responsible for checking whether the user has access to the component or not. Rather a wrapper component should be created to ensure code reusability. Has it not only been Settings.react.js and 10 more components, would you like to add this chunk of code in all those 10 components? I guess not, so I guess creating a separate protected route component would be better. @plxity